src/test/java/org/broadinstitute/hellbender/engine/GATKToolUnitTest.java

```diff
@@ -141,6 +157,120 @@ public void traverse() {
+
+        clp.parseArguments(System.out, args);
+
+        // This method would throw if sequence dictionary validation failed.
+        // Here we are testing that it throws when the input arguments are incompatible,
+        // and that it functions normally when the arguments are correct.
+        tool.onStartup();
+    }
+
+    @Test(dataProvider="sequenceDictionaryTestValues")
+    public void TestGATKToolWithSequenceDictionaryOk(String masterSequenceDictionaryFile,
+                                                     String otherSeqArg, String otherSequenceFile,
+                                                     String cramRefArg, String cramRefFile) throws Exception {
+
+        testGATKToolWithSequenceDictionaryHelper(masterSequenceDictionaryFile,
+                                                 otherSeqArg,
+                                                 otherSequenceFile,
+                                                 cramRefArg,
+                                                 cramRefFile);
+    }
+
+    @Test(dataProvider="sequenceDictionaryTestValuesIncompatible",
+            expectedExceptions = UserException.IncompatibleSequenceDictionaries.class)
+    public void TestGATKToolWithSequenceDictionaryException(String masterSequenceDictionaryFile,
+                                                            String otherSeqArg, String otherSequenceFile,
+                                                            String cramRefArg, String cramRefFile) throws Exception {
+
+        testGATKToolWithSequenceDictionaryHelper(masterSequenceDictionaryFile,
+                                                 otherSeqArg,
+                                                 otherSequenceFile,
+                                                 cramRefArg,
+                                                 cramRefFile);
+    }
+
+    @DataProvider
+    public Object[][] provideMultipleSequences() {
+
+        final String dict1FileName = ReferenceUtils.getFastaDictionaryFileName(v37_chr17_1Mb_Reference);
+        final SAMSequenceDictionary dict1 = ReferenceUtils.loadFastaDictionary(new File(dict1FileName));
+
+        final String dict2FileName = ReferenceUtils.getFastaDictionaryFileName(b37_reference_20_21);
+        final SAMSequenceDictionary dict2 = ReferenceUtils.loadFastaDictionary(new File(dict2FileName));
+
+        final String dict3FileName = ReferenceUtils.getFastaDictionaryFileName(hg19MiniReference);
+        final SAMSequenceDictionary dict3 = ReferenceUtils.loadFastaDictionary(new File(dict3FileName));
+
+        return new Object[][] {
+                { dict1FileName, "--input", NA12878_chr17_1k_BAM, dict1, null, null },
+                { dict2FileName, "--input", BQSR_WGS_B37_CH20_21_10M_100_CRAM, dict2, "--reference", b37_reference_20_21 },
+                { dict3FileName, "--reference", hg19MicroReference, dict3, null, null },
```

Comment:
Can you confirm that in at least some of these test cases, the dictionary being passed in as the master is not exactly equal to the dictionary from the other source(s)?